src/arrayMethodJoin.js

```diff
@@ -4,8 +4,30 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
+  [].__proto__.join2 = function(separator = ',') {
     // write code here
+    let result = '';
+    let separ = separator;
+
+    if (separator === null) {
+      separ = 'null';
+    }
+
+    for (let i = 0; i < this.length; i++) {
+      let wordToAdd = this[i];
+
+      if (this[i] === null || this[i] === undefined) {
+        wordToAdd = '';
+      }
```

Comment:
Let's think in the opposite way:
if not null or undefined - you should add word + separator, if not - do nothing